notation login CLI

SteveLasker opened the pull request:
Updates for #119 
Signed-off-by: Steve Lasker <stevenlasker@hotmail.com>

shizhMSFT commented:
/cc @binbin-li for comments

shizhMSFT commented on specs/notation-cli.md:
```diff
@@ -127,6 +129,32 @@ OPTIONS:
+   notation login [options] [server]
+
+OPTIONS:
+   --username value, -u value    Username for registry operations
+   --password value, -p value    Password for registry operations
```
How about environment variables associated with `-u` and `-p` like `NOTATION_USERNAME` and `NOTATION_PASSWORD`?

gokarnm replied:
```suggestion
   --username value, -u value    Username for registry authentication
   --password value, -p value    Password for registry authentication
```

gokarnm replied:
Should we also support `--password-stdin` like [docker cli](https://docs.docker.com/engine/reference/commandline/login/#provide-a-password-using-stdin) does?

shizhMSFT reviewed (approved):
LGTM with suggestions.

gokarnm commented on specs/notation-cli.md:
````diff
@@ -127,6 +129,32 @@ OPTIONS:
+```console
+notation login --help
+NAME:
+   notation login - Provides credentials for authenticated registry operations
````
```suggestion
   notation login - Login to a registry with credentials for authenticated registry operations
```

gokarnm commented on specs/notation-cli.md:
```diff
@@ -36,6 +37,7 @@ COMMANDS:
    certificate, cert  Manage certificates used for verification
    key                Manage keys used for signing
    list, ls           List signatures from remote
+   login              Provide credentials for authenticated registry operations   
```
```suggestion
   login              Login to a registry with credentials for authenticated registry operations
```

SteveLasker replied:
Saying login, where the parameter says login feels redundant. I was just trying to find a synonym verb.

gokarnm commented on specs/notation-cli.md:
```diff
@@ -127,6 +129,32 @@ OPTIONS:
+  <server>                       The registry URL for authentication
+
+GLOBAL ARGUMENTS
+   --plain-http                  Registry access via plain HTTP (default: false)
```
@shizhMSFT, what is `--insecure` used for, and how common is the usage for `--ca-file` (I assume provides custom TLS trust store).

shizhMSFT replied:
`--insecure` is to skip the TLS cert check. Since `notation` is security oriented, we should not provide this functionality.

`--ca-file` is to specify the root CA cert to registry access. See https://github.com/oras-project/oras/issues/217

SteveLasker replied:
@shizhMSFT, I was thinking this would be done with the `insecure registries` option in config.  
Does that still exist? 
Seems calling `--insecure` is too tedious. For the scenarios/instances where https isn't supported, such as IOT environments, I just assumed the config would solve the problem and not something that's done ad-hoc requiring the cli to support it.

gokarnm reviewed (approved):
LGTM with suggestions.

SteveLasker commented:
I've updated the PR to:
- add $NOTATION_USERNAME, $NOTATION_PASSWORD on `notation login`
- Removed -u -p from other commands, streamlinging so users can benefit from existing cred stores, not having to provide -u -p on every command. This is similar to `docker push|pull` where the user either does `docker login`, or benefits from a previously configured store.
